File: Kotlin/프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.kt
class Solution {
    fun solution(ingredient: IntArray): Int {
        var answer: Int = 0

        val list = mutableListOf<Int>()

        for (ing in ingredient) {
            list.add(ing)

            if (list.size >= 4) {
                val size = list.size
                if (list[size - 4] == 1 &&
                    list[size - 3] == 2 &&
                    list[size - 2] == 3 &&
                    list[size - 1] == 1
                ) {
                    answer++

                    list.removeAt(size - 4)
                    list.removeAt(size - 4)
                    list.removeAt(size - 4)
                    list.removeAt(size - 4)
                }
            }
        }


        return answer
    }
}

File: C#/프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.cs
using System;
using System.Text;
using System.Collections.Generic;

public class Solution {
    public int solution(int[] ingredient) {
        int answer = 0;
        Stack<int> stack = new Stack<int>();
        
        foreach(int i in ingredient)
        {
            stack.Push(i);
            
            if(stack.Count >= 4)
            {
                int l = stack.Pop();
                int t = stack.Pop();
                int s = stack.Pop();
                int f = stack.Pop();
                
                if(f == 1 && s == 2 && t == 3 && l == 1)
                {
                    answer++;
                    continue;
                }
                
                stack.Push(f);
                stack.Push(s);
                stack.Push(t);
                stack.Push(l);
            }
        }
        
        return answer;
    }
}